apps/openmw/mwrender/objects.cpp

```diff
@@ -138,6 +138,14 @@ bool Objects::removeObject (const MWWorld::Ptr& ptr)
     return false;
 }
 
+void Objects::updateEffects(float duration)
+{
+    for(PtrAnimationMap::iterator iter = mObjects.begin();iter != mObjects.end();)
```

Comment:
We should really avoid iterating all objects every frame because there are thousands of them in a normal scene. I remember we had a performance issue caused by the same thing in PhysicsSystem::animateCollisionShapes and it was fixed by tracking animated shapes in a separate container.
Usually the osg::UpdateCallback is a very nice way to do updates every frame. Maybe the updateEffects should be ported to an UpdateCallback?